src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.WriteCore.cs

```diff
@@ -11,7 +11,13 @@ public partial class JsonConverter<T>
             JsonSerializerOptions options,
             ref WriteStack state)
         {
-            if (IsValueType)
+            if (
+#if NET5_0_OR_GREATER
+                // Short-circuit the check against "is not null"; treated as a constant by recent versions of the JIT.
+                typeof(T).IsValueType)
```

Comment:
Nice to see this optimization also applied here. Are there any more cases of `IsValueType` + generics where we can do this?